cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/condition/StageStatistics.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api.condition;
+
+/**
+ * Represents statistics associated with the each stage in the pipeline.
+ */
+public interface StageStatistics {
+  /**
+   * @return the number of records given as an input to the stage or {@code null} if the stage is not executed
```

Comment:
or maybe this could have a state that can be used to check if the stage has run already.